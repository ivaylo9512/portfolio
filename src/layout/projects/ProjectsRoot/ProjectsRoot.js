import Background from '@projects/components/Background/Background'
import AnimationsRoot from '@projects/components/AnimationsRoot/AnimationsRoot'
import './ProjectsRoot.scss'
import { useSelector } from 'react-redux'
import { getCurrentSlide } from 'app/slices/animationsSlice'

const ProjectsRoot = () => {
    const currentSlide = useSelector(getCurrentSlide);
    
    return(
        <section data-testid='projects-root' className={`projects-root ${currentSlide}-slide`}>
            <div className='scroll'>
                <Background />
                <AnimationsRoot />
            </div>
        </section>
    )
}

export default ProjectsRoot