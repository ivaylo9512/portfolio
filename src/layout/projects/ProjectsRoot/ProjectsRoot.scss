.projects-root{
    position: relative;
    height: 100vh;
    box-sizing: border-box;
    width: 50%;

    .scroll{
        position: relative;
        height: 100%;
        display: inline-block;
        width: 100%;
        transition: transform 1.5s;
    }

    @media only screen and (min-aspect-ratio: 10 / 6 ) {
        overflow: hidden;
        
        &.second-slide{
            .scroll{
                transform: translateX(calc(-100vw - 64vh));
            }
        }

        &.third-slide{
            .scroll{
                transform: translateX(calc(-200vw + 20vh));
            }
        }

        .scroll{
            width: auto;
        }
    }
    @media only screen and (max-aspect-ratio: 10 / 6) and (min-aspect-ratio: 6 / 11){
        .scroll{
            width: 100%;
        }
    }
}

.projects-section{
    .projects-root{
        @media only screen and (max-aspect-ratio: 10 / 6 ) {
            height: 100%;
            overflow: visible;
        }
    }

}