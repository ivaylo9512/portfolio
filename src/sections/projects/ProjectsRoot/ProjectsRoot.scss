.projects-root{
    position: relative;
    height: 100%;
    box-sizing: border-box;

    .scroll{
        position: relative;
        height: 100%;
        display: inline-block;
    }

    @media only screen and (orientation: landscape) {
        overflow-x: scroll;
    }
}