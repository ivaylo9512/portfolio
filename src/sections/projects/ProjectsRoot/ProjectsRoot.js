import Background from "@projects/components/Background/Background"
import AnimationsRoot from "@projects/components/AnimationsRoot/AnimationsRoot"
import "./ProjectsRoot.scss"

const ProjectsRoot = () => {
    return(
        <section className="projects-root">
            <div className="scroll">
                <Background />
                <AnimationsRoot />
            </div>
        </section>
    )
}

export default ProjectsRoot