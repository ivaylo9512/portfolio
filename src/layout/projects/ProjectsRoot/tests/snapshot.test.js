import { shallow } from 'enzyme';
import ProjectsRoot from '../ProjectsRoot'
import * as Redux from 'react-redux';

describe('ProjectsRoot snapshot tests', () => {
    let selectorSpy;

    beforeEach(() => {
        selectorSpy = jest.spyOn(Redux, 'useSelector');
    })

    const createWrapper = (curretnSlide) => {
        selectorSpy.mockReturnValue(curretnSlide); 
     
        return shallow(
            <ProjectsRoot />
        )
    }

    it('should match snapshot', () => {
        const wrapper = createWrapper('first');

        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot with second slide', () => {
        const wrapper = createWrapper('second');

        expect(wrapper).toMatchSnapshot();
    })
})