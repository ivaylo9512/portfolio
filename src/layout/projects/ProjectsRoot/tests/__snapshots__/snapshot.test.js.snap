// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsRoot snapshot tests should match snapshot 1`] = `
<section
  className="projects-root first-slide"
  data-testid="projects-root"
>
  <div
    className="scroll"
  >
    <Background />
    <Animations />
  </div>
</section>
`;

exports[`ProjectsRoot snapshot tests should match snapshot with second slide 1`] = `
<section
  className="projects-root second-slide"
  data-testid="projects-root"
>
  <div
    className="scroll"
  >
    <Background />
    <Animations />
  </div>
</section>
`;
